docs/query-builder.md

```diff
@@ -448,64 +448,161 @@ DB::collection('items')
 
 **Push**
 
-Add items to an array.
+Add one or multiple values to the items array.
 
 ```php
+// Push the value to the matched documents
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots');
+  ->where('name', 'John')
+  // Push a single value to the items array
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  // Push multiple values to the items array
+  ->push('items', ['hat', 'jeans']);
+// Result:
+// items: ['boots', 'hat', 'jeans']
 
+// Or
+
+// Push the value to an exact document
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add an array as a value to the messages array. value arrays must be nested into a list array, otherwise they are
+interpreted as a list of scalar values.
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an array as a value to the items array
+  ->push('messages', [
+      [
+          'from'    => 'Jane Doe',
+          'message' => 'Hi John',
+      ],
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
 
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots', true);
+  ->where('name', 'John')
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  ->push('items', ['hat', 'boots', 'jeans'], true);
+// Result:
+// items: ['boots', 'hat', 'jeans']
+
+// Or
+
+$user->push('messages', [
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
 
-$user->push('items', 'boots', true);
+$user->push('messages', [
+    [
+        'from'    => 'Jess Doe',
+        'message' => 'Hi',
+    ],
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+], true);
```

Comment:
This example may open a can of worms about how MongoDB compares documents. For instance, if you happened to swap the order of `from` and `messages` here, the document would likely be appended and you'd end up with three elements in the `messages` array. [Comparison and Sort Order: Objects](https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#objects) goes into more detail about that.

If it's not too useful to readers, I'd propose removing it. An earlier example already demonstrates that we can call `push()` on a model object directly, so I wouldn't push for repeating that multiple times (pun intended).

Reply:
> If it's not too useful to readers, I'd propose removing it.

It's the reason that I've opened this PR. Users get confused when they want to use these methods (like this issue #2387). To reduce the complexity here, I suggest using arrays with only one key and value.

```suggestion
$user->push('messages', [
    [ 'message' => 'Hi' ],
    [ 'message' => 'Hi John' ],
], true);
```

Reply:
Ah, just saw your comment in https://github.com/mongodb/laravel-mongodb/issues/2387#issuecomment-1819110263. In that case, I think it's fine to leave the embedded documents as-is with two fields. Putting myself in the reader's shoes for a moment, I think seeing an embedded document with a single string field would make me wonder why the author didn't just store `messages` as a list of strings.